app/forms/candidate_interface/equality_and_diversity/disability_status_form.rb

```diff
@@ -18,18 +18,15 @@ def self.build_from_application(application_form)
     def save(application_form)
       return false unless valid?
 
-      if application_form.equality_and_diversity.nil?
-        application_form.update(equality_and_diversity: {
-          'disabilities' => [],
-          'hesa_disabilities' => [hesa_code].compact,
-        })
-      elsif reset_disabilities?(application_form)
+      application_form.equality_and_diversity = {} if application_form.equality_and_diversity.nil?
+
+      if application_form.equality_and_diversity['disabilities'].nil? || reset_disabilities?(application_form)
```

Comment:
Consider moving `application_form.equality_and_diversity['disabilities'].nil?` to `reset_disabilities?`

Reply:
I just noticed it was actually moved out of there.. I don't know if I'm missing some context here, but is there any reason as to why?

Reply:
Yes, and I did wonder if having one method to answer the conditional would be better but it seemed like there are 2 different cases for how enter this condition:

1. disabilities key is missing from equality and diversity data (when the user has never filled it in)
2. disabilities key is present but the values are being 'reset' (when the user is changing their answers).

Perhaps I should have been bolder and removed the `reset_...` method naming as it's just updating? The tests seem to mention resetting a lot but I wonder if it's the right term?

Reply:
Ok, I see what you mean.